### PR TITLE
Feature email

### PR DESCRIPTION
**Pull request by nikosev**
Add support for parsing email
Fix condition of FullNameParsing class
Reoerder changelog

### CODE REVIEW

File: lib/FullNameParser.php [hunk 1]
```diff
@@ -340,6 +349,21 @@ public function get_pro_suffix($name)
...
+    /**
+     * Function to check name for existence of email address
+     *
+     * @param string $name the name you wish to test against
+     * @return mixed returns email address if exists, false otherwise
+     */
+    protected function get_email($name)
+    {
+        if (preg_match("/[^@\s]*@[^@\s]*\.[^@\s]*/", $name, $matches)) {
+            return $matches[0];
```
Comment (NicolasLiampotis) [inline lib/FullNameParser.php:361]: Is this regex compatible with the regex used by `filter_var($email, FILTER_VALIDATE_EMAIL)`: https://github.com/php/php-src/blob/8aa92c14650fe0dbf856112d805d8e8bfcbc12fd/ext/filter/logical_filters.c#L648
If not we might need to make and additional check using `filter_var` before returning the mathed string.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,21 +5,29 @@ All notable changes to this project will be documented in this file.
...
+- Support for parsing email address
 
-- FullNameParser library
-  - PHP library to split names into their respective components (first, last, etc)
-- FullNameParsing class
-  - Module class that uses the library
+### Fixed
 
-## [v1.0.1]
+- Fix attribute names in the condition of FullNameParsing class
```
Comment (NicolasLiampotis) [inline CHANGELOG.md]: Replace with 'Fix check of existing first/last name information in user attributes based on configured attribute names'

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: lib/FullNameParser.php (hunk 1 above)
Comment (NicolasLiampotis) [reply lib/FullNameParser.php:361]: or better use https://php.net/manual/en/function.mailparse-rfc822-parse-addresses.php

File: CHANGELOG.md (hunk 2 above)
Comment (nikosev) [reply CHANGELOG.md]: Done

File: lib/FullNameParser.php (hunk 1 above)
Comment (NicolasLiampotis) [reply lib/FullNameParser.php:361]: Need to revisit this in a next version

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**